src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -43,34 +43,34 @@ abstract class Particle : WasmObject() {
     open fun onHandleSync(handle: Handle, allSynced: Boolean) {}
 
     fun renderOutput() {
-      log("renderOutput")
-      val slotName = ""
-      val template = getTemplate(slotName)
-      val dict = populateModel(slotName)
-      val model = StringEncoder.encodeDictionary(dict)
-      onRenderOutput(toWasmAddress(), template.toWasmString(), model.toWasmString())
+        log("renderOutput")
+        val slotName = ""
+        val template = getTemplate(slotName)
+        val dict = populateModel(slotName)
+        val model = StringEncoder.encodeDictionary(dict)
```

Comment:
Totally optional nit: might be worth creating an extension function for `Map<String, String>` to encode it.

```kotlin
fun Map<String, String>.encode(): String = StringEncoder.encodeDictionary(this)
```

Reply:
This is a good idea, but string encoding / decoding of dictionaries is an area with a lot of expected change. It seems best to hold out on this until it gets to be a bit more stable.